### PR TITLE
Adds fee, deposit and simulate to transactions

### PR DESCRIPTION
**Pull request by leonardocustodio**
- [x] Calculate deposit
- [x] Calculate fee
- [x] Add simulate arg to mutations
- [x] Add new tests for the mutations
- [x] Add unit tests for the calculations
- [x] Clean the code up

### CODE REVIEW

File: src/GraphQL/Schemas/Primary/Substrate/Mutations/ApproveCollectionMutation.php [hunk 1]
```diff
@@ -96,6 +99,7 @@ public function resolve(
                 'method' => $this->getMutationName(),
                 'encoded_data' => $encodedData,
                 'idempotency_key' => $args['idempotencyKey'] ?? Str::uuid()->toString(),
+                'simulate' => $args['simulate'] ?? false,
```
Comment (v16Studios) [inline src/GraphQL/Schemas/Primary/Substrate/Mutations/ApproveCollectionMutation.php:102]: Do we need a null coalesce if the arg has a defaultValue?  I believe it should always be set, even if not passed directly.
```suggestion
                'simulate' => $args['simulate'],
```

Comment (leonardocustodio) [reply src/GraphQL/Schemas/Primary/Substrate/Mutations/ApproveCollectionMutation.php:102]: Thanks, removed

File: database/migrations/2023_08_16_184438_add_fee_to_transactions_table.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class () extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('transactions', function (Blueprint $table) {
```
Comment (enjinabner) [inline database/migrations/2023_08_16_184438_add_fee_to_transactions_table.php:13]: You'll need to update the schema dump in `database\schema\mysql-schema.sql` so unit tests will not fail.
You can do this by running `php vendor/bin/testbench schema:dump` and copy the dump from  `vendor/orchestra/testbench-core/laravel/database/schema`

File: src/GraphQL/Types/Substrate/TransactionType.php [hunk 3]
```diff
@@ -67,6 +68,17 @@ public function fields(): array
                 'description' => __('enjin-platform::type.transaction.field.encodedData'),
                 'alias' => 'encoded_data',
             ],
+            'fee' => [
+                'type' => GraphQL::type('BigInt'),
+                'description' => __('enjin-platform::type.transaction.field.fee'),
```
Comment (enjinabner) [inline src/GraphQL/Types/Substrate/TransactionType.php:73]: I don't see translation file being updated

File: src/GraphQL/Types/Substrate/TransactionType.php [hunk 4]
```diff
@@ -67,6 +68,17 @@ public function fields(): array
...
+                'resolve' => function ($transaction) {
+                    return isset($transaction['id']) ? Arr::get($transaction, 'fee') : Substrate::getFee($transaction['encoded_data']);
+                },
+            ],
+            'deposit' => [
+                'type' => GraphQL::type('BigInt'),
+                'description' => __('enjin-platform::type.transaction.field.deposit'),
```
Comment (enjinabner) [inline src/GraphQL/Types/Substrate/TransactionType.php:80]: I don't see translation file being updated

File: src/GraphQL/Types/Input/Substrate/Traits/HasSimulateField.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /**
+     * Get the idempotency field.
+     */
+    public function getSimulateField(
+        ?string $simulateDesc = null,
+    ): array {
+        $simulateType = [
+            'type' => GraphQL::type('Boolean'),
+            'description' => $simulateDesc ?: __('enjin-platform::args.simulate'),
```
Comment (enjinabner) [inline src/GraphQL/Types/Input/Substrate/Traits/HasSimulateField.php:17]: I don't see translation file being updated

**Review by enjinabner — CHANGES_REQUESTED** [review]

File: database/migrations/2023_08_16_184438_add_fee_to_transactions_table.php (hunk 2 above)
Comment (leonardocustodio) [reply database/migrations/2023_08_16_184438_add_fee_to_transactions_table.php:13]: Nice

**Review by enjinabner — APPROVED** [review]

**Pull request merged**